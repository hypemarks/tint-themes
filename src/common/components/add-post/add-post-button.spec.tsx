import { render } from '@testing-library/react';
import React from 'react';
import AddPostButton from './add-post-button';

const addPostConfig = {
  addPostText: 'title',
  addPostStyles: {},
  addPost: jest.fn(),
};

describe('Test <AddPost /> Component', () => {
  it('should render snapshot', () => {
    const wrapper = render(
      <AddPostButton
        text={addPostConfig.addPostText}
        style={addPostConfig.addPostStyles}
        onClick={addPostConfig.addPost}
      />
    );
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
  it('should render add-post with text content equal title', () => {
    const wrapper = render(
      <AddPostButton
        text={addPostConfig.addPostText}
        style={addPostConfig.addPostStyles}
        onClick={addPostConfig.addPost}
      />
    );
    const addPost = wrapper.queryByTestId('add-post');
    expect(addPost).toHaveTextContent(addPostConfig.addPostText);
    expect(addPost).toBeInTheDocument();
  });

  it('should render add-post with specific inline style color', () => {
    const btnColor = '#f2f2f2';
    addPostConfig.addPostStyles = {
      color: btnColor,
    };
    const wrapper = render(
      <AddPostButton
        text={addPostConfig.addPostText}
        style={addPostConfig.addPostStyles}
        onClick={addPostConfig.addPost}
      />
    );
    const addPost = wrapper.queryByTestId('add-post');
    expect(addPost).toHaveStyle(`color: ${btnColor}`);
  });

  it('should render add-post with specific inline style background', () => {
    const btnColor = '#f2f2f2';
    addPostConfig.addPostStyles = {
      background: btnColor,
    };
    const wrapper = render(
      <AddPostButton
        text={addPostConfig.addPostText}
        style={addPostConfig.addPostStyles}
        onClick={addPostConfig.addPost}
      />
    );
    const addPost = wrapper.queryByTestId('add-post');
    expect(addPost).toHaveStyle(`background: ${btnColor}`);
  });
});
