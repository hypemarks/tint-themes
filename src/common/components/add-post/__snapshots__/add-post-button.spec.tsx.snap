// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AddPost /> Component should render snapshot 1`] = `
<DocumentFragment>
  <button
    class="add-post button"
    data-testid="add-post"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-plus-square fa-w-14 fa-sm "
      data-icon="plus-square"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
        fill="currentColor"
      />
    </svg>
    title
  </button>
</DocumentFragment>
`;
